tests/petsc/block_matrices_01.cc

```diff
@@ -91,6 +91,88 @@ test()
                  ExcInternalError());
         }
     }
+
+  // Extract the PETSc MATNEST, create an array of PETSc matrices and assign
+  // to a new BlockSparseMatrix
+  std::array<Mat, 2> arrayRows0 = {
+    {tmp2.block(0, 0).petsc_matrix(), tmp2.block(0, 1).petsc_matrix()}};
+  std::array<Mat, 2> arrayRows1 = {
+    {tmp2.block(1, 0).petsc_matrix(), tmp2.block(1, 1).petsc_matrix()}};
+  std::array<std::array<Mat, 2>, 2> arrayMat = {{arrayRows0, arrayRows1}};
```

Comment:
This is more of a style issue: You tend to add new tests to existing ones. We typically just clone an existing test and replace its content by a new one. In other words, our tests tend to check a single concept and are more fine grained. Unless there is a good reason, my preference would be to stay with our style, rather than adding to existing tests.

Reply:
because in my experience `almost unit-tests` catch many more API coding issues than `strictly unit-tests`. Here for example I'm passing around PETSc `Mat`s from one Deal.II object to another, testing object referencing.

I understand this style of testing is more in line with software having abstract class APIs like PETSc (e.g. `Vec`, `Mat`, ...) then Deal.II, but my hope is that also Deal.II will make a move on having a few abstract classes in key places like linear algebra.

Reply:
I didn't intend to have a philosophical discussion :-) I just wanted to point out that historically we've done it differently and that that's just the style generally used by the ~7000 existing tests ;-)